pyo3-macros-backend/src/utils.rs

```diff
@@ -41,6 +41,19 @@ pub fn is_python(ty: &syn::Type) -> bool {
     }
 }
 
+/// Check if the given type `ty` is `pyo3::coroutine::CoroutineCancel`.
+pub fn is_coroutine_cancel(ty: &syn::Type) -> bool {
+    match unwrap_ty_group(ty) {
+        syn::Type::Path(typath) => typath
+            .path
+            .segments
+            .last()
+            .map(|seg| seg.ident == "CoroutineCancel")
```

Comment:
From experience with these proc macros we've generally found that having to rely on text idents for making user-facing behavioural changes is not great. (e.g. what if the user happens to make their own type called `CoroutineCancel`? We can't distinguish that. Also this functionality breaks if the user chooses to rename the import.)`

I know we do it for `py: Python` and also `Option`; I'm hoping one day we remove the special-casing on `Option`. The `py: Python` is so common that it's a reasonably clear ergonomic win but I would remove the text-based ident detection in a heartbeat.

An alternative way to do this would be to have `#[pyo3(cancel_handle = "x")]` which then makes `x` get treated as this special argument.

Reply:
I agree. I started in this direction, but I finally chose the fast and dirty way to show you the feature first.

I will add a an attribute.

Reply:
To be honest, I feel like this PR is repeatedly derailing as new stuff is added on top after we tried to whittle down the scope to increase the chances of this getting a proper review.

Is the cancellation part really necessary for an MVP? Could it be a follow-up PR?

Reply:
I would also prefer cancellation to be split out 👍 (as implied but not insisted upon in a previous comment)

Reply:
Cancellation is not necessary in the MVP, I will revert it. Sorry for spreading myself too thin, I should use pyo3-async to make these kind of design attempts (that's its primary goal)